src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -596,12 +595,15 @@ private static bool IsRightCountComparison(IBinaryOperation binaryOperation, out
             return true;
         }
 
-        private static bool TypeContainsVisibleProperty(OperationAnalysisContext context, ITypeSymbol type, string propertyName, SpecialType propertyType, out ISymbol? propertySymbol)
-            => TypeContainsVisibleProperty(context, type, propertyName, propertyType, propertyType, out propertySymbol);
+        private static bool TypeContainsVisiblePropertyWithoutCircularity(OperationAnalysisContext context, IOperation operation, ITypeSymbol type,
+            string propertyName, SpecialType propertyType, [NotNullWhen(true)] out ISymbol? propertySymbol)
+            => TypeContainsVisiblePropertyWithoutCircularity(context, operation, type, propertyName, propertyType, propertyType, out propertySymbol);
 
-        private static bool TypeContainsVisibleProperty(OperationAnalysisContext context, ITypeSymbol type, string propertyName, SpecialType lowerBound, SpecialType upperBound, out ISymbol? propertySymbol)
+        private static bool TypeContainsVisiblePropertyWithoutCircularity(OperationAnalysisContext context, IOperation operation, ITypeSymbol type,
+            string propertyName, SpecialType lowerBound, SpecialType upperBound, [NotNullWhen(true)] out ISymbol? propertySymbol)
         {
-            if (TypeContainsMember(context, type, propertyName, lowerBound, upperBound, out bool isPropertyValidAndVisible, out propertySymbol!))
+            if (TypeContainsMember(context, type, propertyName, lowerBound, upperBound, out bool isPropertyValidAndVisible, out propertySymbol) &&
+                !IsPropertyGetterUsingThisInstance(context, operation, propertySymbol))
             {
                 return isPropertyValidAndVisible;
```

Comment:
You don't have to call `IsPropertyGetterUsingThisInstance` when `isPropertyValidAndVisible = false`, right? The same would apply to the other calls.
```suggestion
            if (TypeContainsMember(context, type, propertyName, lowerBound, upperBound, out bool isPropertyValidAndVisible, out propertySymbol)
            {
                return isPropertyValidAndVisible && !IsPropertyGetterUsingThisInstance(context, operation, propertySymbol));
```